### ISSUE TITLE
After the ESPhome 2023.12.0 update, the ESP reboots in a loop.

### ISSUE THREAD

**Issue by WarC0zes** [landonr/lilygo-tdisplays3-esphome#49]
Hi,
the compile worked, but ESP rebooted in a loop. I restore to version 2023.11.6, it's good ESP works.
I did a "clean build files" and recompiled, it's the same. I can't see the logs because I can't connect to the device.

I've the ESP32-S3 T-Display model without touch.

```
esphome:
  name: s3-display
  friendly_name: ESP32-S3 T-Display
  platformio_options:
    board_build.f_flash: 80000000L
    board_build.partitions: default_16MB.csv
    board_build.arduino.memory_type: qio_opi

external_components:
  - source: github://landonr/lilygo-tdisplays3-esphome
    components: [tdisplays3]

esp32:
  board: esp32-s3-devkitc-1
  variant: esp32s3
  framework:
    type: arduino
  flash_size: 16MB

# Enable logging
logger:

# Enable Home Assistant API
api:
  encryption:
    key: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"

ota:
  password: "xxxxxxxxxxxxxxxxxxxxxxxxx"

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
  manual_ip:
    static_ip: 192.168.1.xx
    gateway: !secret wifi_gtw
    subnet: !secret wifi_sub
    dns1: !secret wifi_dns1

web_server:

uart:
  tx_pin: 17
  rx_pin: 18
  baud_rate: 256000
  id: uart_1
  parity: NONE
  stop_bits: 1

ld2410:
  - uart_id: uart_1

time:
  - platform: homeassistant
    id: ha_time

sensor:
  - platform: homeassistant
    id: temp_1
    entity_id: sensor.atc_20a2_temperature

  - platform: homeassistant
    id: temp_2
    entity_id: sensor.atc_67c6_temperature

  - platform: homeassistant
    id: power
    entity_id: sensor.0xa4c1383a5eeb5542_power

  - platform: ld2410
    light: #intensité lumineuse
      name: light
    moving_distance: #distance cible en mouvement
      name : Moving Distance
    still_distance: #distance cible immobile
      name: Still Distance
    moving_energy: #quantité de mouvement
      name: Move Energy
    still_energy: #quantité d'immobilité ??
      name: Still Energy
    detection_distance: #distance de détection
      name: Detection Distance

binary_sensor:
  - platform: gpio
    pin: 
      number: GPIO0
      inverted: true
    name: "Button 1"
    on_press:
      then:
        - light.toggle: light_s3

  - platform: gpio
    pin: 
      number: GPIO14
      inverted: true
      mode:
         input: true
    name: "Button 2"
    filters:
    - delayed_on: 10ms
    on_press: 
      - lambda:  |-
          if (id(disp).get_active_page() == id(page1)) {
            ESP_LOGD("custom", "Switch to Page 2");
            id(disp).show_page(id(page2));
          } 
          else if (id(disp).get_active_page() == id(page2)) {
            ESP_LOGD("custom", "Switch to Page 1");
            id(disp).show_page(id(page1));
          }  
          else {
            // Might not need this but here for future
          }

  - platform: status
    name: "Statut"
    id: system_status
    entity_category: "diagnostic"

  - platform: homeassistant
    entity_id: binary_sensor.0x00158d0009439ec9_contact
    id: door_frigo

  - platform: ld2410
    has_target: #Présence
      name: Presence
    has_moving_target: #Cible mobile
      name: Moving Target
    has_still_target: #Cible immobile
      name: Still Target

switch:
  - platform: restart
    name: "Restart"
    entity_category: config

  - platform: ld2410
    engineering_mode:
      name: "engineering mode"

button:
  - platform: ld2410
    restart:
      name: "ld2410 restart"

text_sensor:
  - platform: template
    id: s3_uptime
    name: "Uptime"
    lambda: |-
      int seconds = (id(uptime_sec).state);
      int days = seconds / (24 * 3600);
      seconds = seconds % (24 * 3600); 
      int hours = seconds / 3600;
      seconds = seconds % 3600;
      int minutes = seconds /  60;
      seconds = seconds % 60;
      if ( days ) {
        return { (String(days) +"d " + String(hours) +"h " + String(minutes) +"m "+ String(seconds) +"s ").c_str() };
      } else if ( hours ) {
        return { (String(hours) +"h " + String(minutes) +"m "+ String(seconds) +"s ").c_str() };
      } else if ( minutes ) {
        return { (String(minutes) +"m "+ String(seconds) +"s ").c_str() };
      } else {
        return { (String(seconds) +"s ").c_str() };
      }
    icon: mdi:clock-start
    update_interval: 60s
    entity_category: "diagnostic"

number:
  - platform: ld2410
    timeout: #Temps pour passer le binary sensor Présence à faux
      name: timeout
    light_threshold: #Seuil de lumière
      name: light threshold
    max_move_distance_gate: #Distance de détection du mouvemenet 2-8
      name: max move distance gate
    max_still_distance_gate: #Distance de détection d'immobilité 2-8
      name: max still distance gate

output:
  - platform: ledc
    pin: GPIO38
    id: gpio38
    frequency: 2000

light:
  - platform: monochromatic
    output: gpio38
    name: "Backlight"
    id: light_s3
    restore_mode: RESTORE_DEFAULT_ON

font:
  - file: "gfonts://Roboto"
    id: roboto
    size: 30

  - file: "gfonts://Roboto"
    id: roboto_12
    size: 12

  - file: "gfonts://Roboto"
    id: roboto_20
    size: 18

  - file: 'fonts/materialdesignicons-webfont.ttf'
    id: icon_font
    size: 58
    glyphs: [
      "\U000F081B", # mdi-door-closed
      "\U000F081C", # mdi-door-open
      ]

  - file: 'fonts/materialdesignicons-webfont.ttf'
    id: icon_font_po
    size: 55
    glyphs: [
      "\U000F0241", # mdi-flash
      "\U000F0243", # mdi-flash-off
      ]

  - file: 'fonts/materialdesignicons-webfont.ttf'
    id: icon_meteo
    size: 58
    glyphs: [
      "\U000F0594", # mdi-weather-night
      "\U000F0590", # mdi-weather-cloudy
      "\U000F0591", # mdi-weather-fog
      "\U000F0592", # mdi-weather-hail
      "\U000F0593", # mdi-weather-lightning
      "\U000F067E", # mdi-weather-lightning-rainy
      "\U000F0595", # mdi-weather-partly-cloudy
      "\U000F0596", # mdi-weather-pouring
      "\U000F0597", # mdi-weather-rainy
      "\U000F0598", # mdi-weather-snowy
      "\U000F067F", # mdi-weather-snowy-rainy
      "\U000F0599", # mdi-weather-sunny
      "\U000F059D", # mdi-weather-windy
      "\U000F059E", # mdi-weather-windy-variant
      ]

display:
  - platform: tdisplays3
    id: disp
    rotation: 90    
    update_interval: 1s
    pages:
      - id: page1
        lambda: |-
          it.printf(40, 1, id(roboto), Color(0, 0, 255), id(ha_time).now().strftime("%H:%M:%S %d/%m/%y").c_str());
          it.print(1,  40, id(roboto), Color(255, 255, 255), TextAlign::LEFT, "Congelateur:");  
          it.print(1,  80, id(roboto), Color(255, 255, 255), TextAlign::LEFT, "Refrigerateur:");
          it.printf(191, 40, id(roboto), Color(0, 255, 0), "%.1f°C", id(temp_1).state);
          it.printf(216, 80, id(roboto), Color(0, 255, 0), "%.1f°C", id(temp_2).state);
          it.print(1, 122, id(roboto), Color(255, 255, 255), TextAlign::LEFT, "Porte:");
          if (id(door_frigo).state == false) {
            it.print(88, 122, id(roboto), Color(0, 255, 0), TextAlign::LEFT, "Fermer");
          } else {
            it.print(88, 122, id(roboto), Color(255, 0, 0), TextAlign::LEFT, "Ouverte");
          }
          if (id(door_frigo).state == false) {
            it.printf(200, 112, id(icon_font), Color(0, 255, 0), TextAlign::LEFT, "\U000F081B");
          } else {
            it.printf(200, 112, id(icon_font), Color(255, 0, 0), TextAlign::LEFT, "\U000F081C");
          }
          if (id(power).state >= 1) {
            it.printf(260, 115, id(icon_font_po), Color(255, 0, 0), TextAlign::LEFT, "\U000F0241");
          } else {
            it.printf(260, 115, id(icon_font_po), Color(0, 255, 0), TextAlign::LEFT, "\U000F0243");
          }
      - id: page2
        lambda: |-
          it.rectangle(0,  0, it.get_width(), it.get_height(), Color(0, 0, 255));
          it.rectangle(0, 20, it.get_width(), it.get_height(), Color(0, 0, 255));
          it.print(8, 5, id(roboto_12), Color(255, 255, 0), TextAlign::LEFT, "Today");
          it.printf(85, 5, id(roboto_12), Color(255, 255, 0), TextAlign::LEFT, "%.0fh", id(weather_heure_h0).state);
          it.printf(149, 5, id(roboto_12), Color(255, 255, 0), TextAlign::LEFT, "%.0fh", id(weather_heure_h1).state);
          it.printf(213, 5, id(roboto_12), Color(255, 255, 0), TextAlign::LEFT, "%.0fh", id(weather_heure_h2).state);
          it.print(260, 5, id(roboto_12), Color(255, 255, 0), TextAlign::LEFT, "Tomorrow");
...

```

**Comment by Anto79-ops** [comment]
when I connect to the device using a serial cable and use the ESPHome webpage, it works as it should and able to see the logs, with everything looking normal.

The moment I remove the cable and plug it into a non-data usb, it goes into the loop thing again.

**Comment by guillempages** [comment]
I'm also seeing this issue. I've commented out all display and touchscreen related configurations, and the reboot loop is still happening, so this is not related to the external component in this repository, but a more generic issue with ESPHome.

I think there was some mention on discord about some serial hardware uart needing to be configured with the latest version, but I'm not sure about the details; I'll try to dig it up and test that.

**Comment by guillempages** [comment]
I managed to track the issue to https://github.com/esphome/esphome/pull/4853
As a workaround, you can add this to your `logger` configuration, that way the device does not enter the reboot loop, but you also cannot read the logs via USB.

```
logger:
  hardware_uart: UART0
```

Not a real solution, but a workaround.

**Comment by Anto79-ops** [comment]
hey thanks! I this could workd for me and I'll try it soon. I don't really need the logs via USB on this device.

By the way, for an official fix, is this something coming from this repo or ESPHome? thanks

**Comment by guillempages** [comment]
An official fix should come from the official esphome repo.

You might want to see if there is already an open issue there and create one if it isn't. Unfortunately I'm not at the computer anymore to do that myself 🙁

**Comment by Anto79-ops** [comment]
looks like the fix is coming in 12,2 any moment now....thanks!

**Comment by WarC0zes** [comment]
> looks like the fix is coming in 12,2 any moment now....thanks!

ESP32-S3 and ESP-IDF don't play well with USB_CDC and need USB_SERIAL_JTAG https://github.com/esphome/esphome/pull/5929

good new,  waiting for the release.

**Comment by WarC0zes** [comment]
Fix with ESPHome 2023.12.2.
No more loop problem.
